ros_buildfarm/bloom_status.py

```diff
@@ -76,41 +79,57 @@ def _get_package_info(g, distro_info):
                         headers = {}
                     tracks = _download_yaml(tracks_url, headers).get('tracks', {})
                     cached_tracks[url] = tracks
-                release_tag = tracks.get(distro, {}).get('release_tag', None)
-                if release_tag == ':{ask}' and 'last_release' in tracks[distro]:
-                    pkg[distro]['last_release'] = tracks[distro]['last_release']
+                dtracks = tracks.get(distro, {})
+                m = GITHUB_TOKEN_PATTERN.match(dtracks.get('vcs_uri', ''))
+                if m:
+                    distro_d['org'] = m.group(2)
+                upstream = dtracks.get('devel_branch', None)
+                if upstream:
+                    distro_d['upstream'] = upstream
+                release_tag = dtracks.get('release_tag', None)
+                if release_tag == ':{ask}' and 'last_release' in dtracks:
+                    distro_d['last_release'] = dtracks['last_release']
 
             if 'source' in d:
                 upstream = d['source'].get('version', None)
                 if upstream:
-                    pkg[distro]['upstream'] = upstream
-                m = GITHUB_TOKEN_PATTERN.match(d['source'].get('url', ''))
-                if m:
-                    pkg[distro]['org'] = m.group(2)
+                    if 'upstream' in distro_d:
+                        if upstream != distro_d['upstream']:
+                            print('Package {} does not have matching upstream branches for {}. ({} vs. {})'
```

Comment:
Format strings are awesome too! Consider them in the future for legibility.

`f"Package {name} does not have matching..."`